lib/metrics/peer_count.go

```diff
@@ -16,35 +16,47 @@ type PeerCount struct {
 
 func (r *PeerCount) Process(db *sql.DB, errs *common.MetricErrors, data *types.TelemetryRequest) error {
 	if err := json.Unmarshal(*data.TelemetryData, &r); err != nil {
-		errs.Append(data.Id, fmt.Sprintf("Error decoding peer count: %v", err))
+		errs.Append(data.ID, fmt.Sprintf("Error decoding peer count: %v", err))
 		return err
 	}
 
-	stmt, err := db.Prepare("INSERT INTO peerCount (timestamp, nodeName, nodeKeyUid, peerId, peerCount, statusVersion, createdAt, deviceType) VALUES ($1, $2, $3, $4, $5, $6, $7, $8) RETURNING id;")
+	tx, err := db.BeginTx(context.Background(), nil)
```

Comment:
Ideally the context should be passed down. Maybe this one could be used https://github.com/status-im/telemetry/blob/master/telemetry/server.go#L39 ?